substrate/bin/utils/chain-spec-builder/bin/main.rs

```diff
@@ -17,83 +17,24 @@
 };
 use clap::Parser;
-use node_cli::chain_spec;
-use rand::{distributions::Alphanumeric, rngs::OsRng, Rng};
 use sc_chain_spec::{update_code_in_json_chain_spec, GenericChainSpec};
-use sp_core::{crypto::Ss58Codec, sr25519};
 use staging_chain_spec_builder as chain_spec_builder;
 use std::fs;
 
 fn main() -> Result<(), String> {
 	sp_tracing::try_init_simple();
 
 	let builder = ChainSpecBuilder::parse();
-	#[cfg(build_type = "debug")]
-	if matches!(builder.command, ChainSpecBuilderCmd::Generate(_) | ChainSpecBuilderCmd::New(_)) {
-		println!(
-			"The chain spec builder builds a chain specification that includes a Substrate runtime \
-		 compiled as WASM. To ensure proper functioning of the included runtime compile (or run) \
-		 the chain spec builder binary in `--release` mode.\n",
-		 );
-	}
-
 	let chain_spec_path = builder.chain_spec_path.to_path_buf();
-	let mut write_chain_spec = true;
-
-	let chain_spec_json = match builder.command {
-		ChainSpecBuilderCmd::Generate(GenerateCmd {
-			authorities,
-			nominators,
-			endowed,
-			keystore_path,
-		}) => {
-			let authorities = authorities.max(1);
-			let rand_str = || -> String {
-				OsRng.sample_iter(&Alphanumeric).take(32).map(char::from).collect()
-			};
-
-			let authority_seeds = (0..authorities).map(|_| rand_str()).collect::<Vec<_>>();
-			let nominator_seeds = (0..nominators).map(|_| rand_str()).collect::<Vec<_>>();
-			let endowed_seeds = (0..endowed).map(|_| rand_str()).collect::<Vec<_>>();
-			let sudo_seed = rand_str();
-
-			print_seeds(&authority_seeds, &nominator_seeds, &endowed_seeds, &sudo_seed);
-
-			if let Some(keystore_path) = keystore_path {
-				generate_authority_keys_and_store(&authority_seeds, &keystore_path)?;
-			}
-
-			let nominator_accounts = nominator_seeds
-				.into_iter()
-				.map(|seed| {
-					chain_spec::get_account_id_from_seed::<sr25519::Public>(&seed).to_ss58check()
-				})
-				.collect();
-
-			let endowed_accounts = endowed_seeds
-				.into_iter()
-				.map(|seed| {
-					chain_spec::get_account_id_from_seed::<sr25519::Public>(&seed).to_ss58check()
-				})
-				.collect();
-
-			let sudo_account =
-				chain_spec::get_account_id_from_seed::<sr25519::Public>(&sudo_seed).to_ss58check();
 
-			generate_chain_spec(authority_seeds, nominator_accounts, endowed_accounts, sudo_account)
+	match builder.command {
+		ChainSpecBuilderCmd::Create(cmd) => {
+			let chain_spec_json = generate_chain_spec_for_runtime(&cmd)?;
+			fs::write(chain_spec_path, chain_spec_json).map_err(|err| err.to_string())?;
 		},
-		ChainSpecBuilderCmd::New(NewCmd {
-			authority_seeds,
-			nominator_accounts,
-			endowed_accounts,
-			sudo_account,
-		}) =>
-			generate_chain_spec(authority_seeds, nominator_accounts, endowed_accounts, sudo_account),
-		ChainSpecBuilderCmd::Runtime(cmd) => generate_chain_spec_for_runtime(&cmd),
 		ChainSpecBuilderCmd::Edit(EditCmd {
 			ref input_chain_spec,
 			ref runtime_wasm_path,
```

Comment:
Nit: Should any of these be mandatory? Just doing
chain-spec-builder edit --input-chain-spec chain_spec.json does silently nothing.

Reply:
Seems that two separate commands would be better. How about:
- `update-code`
- `convert-to-raw`
?

Probably that would be better then one `edit` with unclear options.

Reply:
done in: ed84b6b